tests/test_email_domain_checker.py

```
@@ -0,0 +1,88 @@
+    result = check_domain("example.com", cache)
+    assert result == {"blocked": False}
+    assert mock_get.call_count == 1  # Should not make another API call
+
+@patch('email_domain_checker.check_domain')
+def test_check_domains_for_emails(mock_check_domain):
+    # Reset the mock for each test
+    mock_check_domain.reset_mock()
+    mock_check_domain.return_value = {"blocked": True}
+
+    with patch('email_domain_checker.API_KEY', 'test_key'):
+        with patch('email_domain_checker.load_cache') as mock_load_cache:
```

Ruff (0.11.9)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)